YAML/YAML.sublime-syntax

```diff
@@ -599,46 +648,37 @@ contexts:
 
   block-pair:
     - match: \?
-      scope: punctuation.definition.key-value.begin.yaml
+      scope: meta.mapping.yaml punctuation.definition.key-value.begin.yaml
       push:
-        - meta_scope: meta.block-mapping.yaml
+        - meta_content_scope: meta.mapping.key.yaml
         - match: (?=\?) # Empty mapping keys & values are allowed
           pop: true
         - match: ^ *(:)
+          scope: meta.mapping.yaml
           captures:
             1: punctuation.separator.key-value.mapping.yaml
           pop: true
         - match: ':'
-          scope: invalid.illegal.expected-newline.yaml
+          scope: meta.mapping.yaml invalid.illegal.expected-newline.yaml
           pop: true
         - include: block-node
-    # Attempt to match plain-out scalars and highlight as "entity.name.tag",
+    # Attempt to match plain-out scalars and highlight as "meta.mapping.key",
     # if followed by a colon
-    - match: |
-        (?x)
-        (?=
-          {{ns_plain_first_plain_out}}
-          (
-              [^\s:]
-            | : \S
-            | \s+ (?![#\s])
-          )*
-          \s*
-          :
-          (\s|$)
-        )
+    # TODO rewrite with branching
```

Comment:
Were you going to leave this TODO here, or fix it in this PR?

Reply:
That will have to wait for a different PR. While it is certainly something that *could* be done in a similar scope, it's not something that I will find the time for in the forseeable future.